provider/aws.go

```diff
@@ -30,7 +30,30 @@ import (
 )
 
 const (
-	hostedZonePrefix = "/hostedzone/"
+	hostedZonePrefix     = "/hostedzone/"
+	elbHostnameSuffix    = ".elb.amazonaws.com"
+	evaluateTargetHealth = false
+	recordTTL            = 300
+)
+
+var (
+	// see: https://docs.aws.amazon.com/general/latest/gr/rande.html
+	canonicalHostedZones = map[string]string{
+		"us-east-1":      "Z35SXDOTRQ7X7K",
```

Comment:
yes, except this 
`
 +func canonicalHostedZone(hostname string) string {
`
is not needed ;)